### ISSUE TITLE
Pyright has problem with combo of `@property` and `@cache`

### ISSUE THREAD

**Issue by MapleCCC** [microsoft/pyright#2133]
**Describe the bug**
The combo use of `@property` and `@cache` is [backed by the document](https://github.com/python/cpython/blob/main/Doc/library/functools.rst?plain=1#L97) of `functools.cached_property`, so it should be considered a valid use case. However, pyright seems to have some bugs in handling such case.

**To Reproduce**
```python
from functools import cache

class Rectangle:
    def __init__(self, length: int, width: int) -> None:
        self._length = length
        self._width = width

    @property
    @cache
    def area(self) -> int:
        return self._length * self._width

def is_large_rectangle(rec: Rectangle) -> bool:
    return rec.area >= 100

rec = Rectangle(10, 10)
print(is_large_rectangle(rec))
```

Pyright complains that `Operator ">=" not supported for types "_lru_cache_wrapper[int]" and "Literal[100]"`.

**Expected behavior**
There should be no error for the above code.

**VS Code extension or command-line**
Pyright command-line tool 1.1.158

**Comment by erictraut** [comment]
This looks to be a limitation or a bug in the typeshed type definition for `cache`. I think pyright is applying the type information correctly here. It matches the behavior of mypy in this case.

```python
    def cache(__user_function: Callable[..., _T]) -> _lru_cache_wrapper[_T]: ...

class _lru_cache_wrapper(Generic[_T]):
    __wrapped__: Callable[..., _T]
    def __call__(self, *args: Hashable, **kwargs: Hashable) -> _T: ...
    def cache_info(self) -> _CacheInfo: ...
    def cache_clear(self) -> None: ...
```

This definition implies that the resulting cache object is callable, and calling it will return the cached value. But that's not how it works at runtime.

I don't know why `cache` was defined as it currently is in `functools.pyi`. My advice is to file an issue in the typeshed repo and start a discussion there. We frequently pull changes from typeshed into pyright, so if a fix is made there, it will soon be fixed in pyright as well.

**Comment by MapleCCC** [comment]
@erictraut Thank you for your patient reply. To be honest, I am not familiar with the type annotation enough to understand your introspection. If it's convenient, could you consider open an issue in the typeshed library? I really don't know how to describe this complex type annotation thing 😵

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
